recipes/antlr4-cppruntime/all/conanfile.py

```diff
@@ -23,133 +29,129 @@ class Antlr4CppRuntimeConan(ConanFile):
         "shared": False,
         "fPIC": True,
     }
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "pkg_config"
     short_paths = True
 
-    compiler_required_cpp17 = {
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
             "Visual Studio": "16",
+            "msvc": "1923",
             "gcc": "7",
             "clang": "5",
             "apple-clang": "9.1"
-    }
-
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+        }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("utfcpp/3.2.1")
-        if self.settings.os in ("FreeBSD", "Linux"):
+        # As of 4.11, antlr4-cppruntime no longer requires libuuid.
+        # Reference: [C++] Remove libuuid dependency (https://github.com/antlr/antlr4/pull/3787)
+        if self.settings.os in ("FreeBSD", "Linux") and Version(self.version) < "4.11":
             self.requires("libuuid/1.0.3")
 
     def validate(self):
-        if str(self.settings.arch).startswith("arm"):
-            raise ConanInvalidConfiguration("arm architectures are not supported")
+        if str(self.info.settings.arch).startswith("arm") and Version(self.version) < "4.11":
+            raise ConanInvalidConfiguration("arm architectures are not supported due to libuuid; please upgrade to antlr 4.11 or later.")
             # Need to deal with missing libuuid on Arm.
             # So far ANTLR delivers macOS binary package.
 
-        compiler = self.settings.compiler
-        compiler_version = tools.Version(self.settings.compiler.version)
-        antlr_version = tools.Version(self.version)
-
-        if compiler == "Visual Studio" and compiler_version < "16":
+        if (self.info.settings.compiler == "Visual Studio" and self.info.settings.compiler.version < "16") or \
+           (self.info.settings.compiler == "msvc" and self.info.settings.compiler.version < "1920"):
```

Comment:
```suggestion
        if (self.info.settings.compiler == "Visual Studio" and Version(self.info.settings.compiler.version) < "16") or \
           (self.info.settings.compiler == "msvc" and Version(self.info.settings.compiler.version) < "1920"):
```

Need to parse version